src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -775,6 +776,58 @@ private void VisitObjectCreationExpressionBase(BoundObjectCreationExpressionBase
                         _diagnostics);
+
+                    if (node.InitializerExpressionOpt is { })
+                    {
+                        // Object initializers are different than a normal constuctor in that the 
+                        // scope of the receiver is determined by evaluating the inputs to the constructor
+                        // *and* all of the initializers. Another way of thinking about this is that
+                        // every argument in an initializer that can escape to the receiver is 
+                        // effectively an argument to the constructor. That means we need to do
+                        // a second mixing pass here where we consider the receiver escaping 
+                        // back into the ref parameters of the constructor.
+                        //
+                        // At the moment this is only a hypothetical problem. Because the language 
+                        // doesn't support ref field of ref struct mixing like this could not actually
+                        // happen in practice. At the same time we want to error on this now so that 
+                        // in a future that we do have ref filed to ref struct this is not a breaking 
+                        // change. Customers can respond to failures like this by putting scoped on
+                        // such parameters in the constructor.
+                        var escapeValues = ArrayBuilder<EscapeValue>.GetInstance();
+                        var escapeTo = GetValEscape(node.InitializerExpressionOpt, _localScopeDepth);
+                        GetEscapeValues(
+                            in methodInfo,
+                            receiver: null,
+                            receiverIsSubjectToCloning: ThreeState.Unknown,
+                            constructor.Parameters,
+                            node.Arguments,
+                            node.ArgumentRefKindsOpt,
+                            node.ArgsToParamsOpt,
+                            ignoreArglistRefKinds: false,
+                            mixableArguments: null,
+                            escapeValues);
+
+                        foreach (var (parameter, argument, _, isRefEscape) in escapeValues)
+                        {
+                            if (!isRefEscape)
+                            {
+                                continue;
+                            }
+
+                            if (parameter?.Type?.IsRefLikeOrAllowsRefLikeType() != true ||
+                                !parameter.RefKind.IsWritableReference())
+                            {
+                                continue;
+                            }
+
+                            if (escapeTo > GetValEscape(argument, _localScopeDepth))
+                            {
+                                Error(_diagnostics, ErrorCode.ERR_CallArgMixing, argument.Syntax, constructor, parameter.Name);
```

Comment:
Should this be a warning in an unsafe region?

Reply:
Pretty sure call argument mixing is always an error. It's not one of the situations that LDM approved for downgrading to warning in `unsafe`.